#! /usr/bin/env python
 
PACKAGE='dbw_mkz_twist_controller'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name			Type		Reconfiguration level 			  Description			Default	 Min Max
gen.add("pub_pedals",	bool_t,	0, "Publish throttle and brake",	True)
gen.add("pub_steering",	bool_t,	0, "Publish steering",				True)

gen.add("speed_kp",	double_t,	0, "Gain",		2.0, 0.0, 100.0)
gen.add("decel_max",	double_t,	0, "Output limit",		3.0, 0.0, 8.0)
gen.add("accel_max",	double_t,	0, "Output limit",		3.0, 0.0, 8.0)
gen.add("max_lat_accel",double_t,	0, "Max lat accel",	8.0, 0.0, 10.0)
gen.add("steer_kp",	double_t,	0, "Gain",		0.0, 0.0, 100.0)
gen.add("wheel_radius",	double_t,	0, "Effective radius", 0.2413, 0.2, 0.2413)
gen.add("vehicle_mass",	double_t,	0, "Curb weight", 1736.35, 1650.0, 1800.0)
gen.add("fuel_capacity",double_t,	0, "Capacity in gal", 13.5, 10.0, 15.0)
gen.add("accel_kp",	double_t,	0, "Gain",		0.4, 0.0, 8.0)
gen.add("accel_ki",	double_t,	0, "Gain",		0.1, 0.0, 8.0)
gen.add("brake_deadband",	double_t,	0, "Deadband",		0.1, 0.0, 1.0)
gen.add("accel_tau",double_t,	0, "LPF time constant",	0.5, 0.0, 10.0)


exit(gen.generate(PACKAGE, PACKAGE, "Controller"))
